### PR TITLE
Add global --debug flag

### PR DESCRIPTION
**Pull request by alinxie**
Added `debug_mode` property in `ContextObject` to be accessed by the `error_and_quit` function in `utils.py`

For Issue: https://github.com/databricks/databricks-cli/issues/110

### CODE REVIEW

**Comment by andrewmchen** [comment]
Can we add a quick test for this?

File: databricks_cli/configure/config.py [hunk 1]
```diff
@@ -58,6 +58,14 @@ def get_profile_from_context():
     return context_object.get_profile()
 
 
+def debug_option(f):
+    def callback(ctx, param, value): #  NOQA
+        context_object = ctx.ensure_object(ContextObject)
+        context_object.set_debug(value)
+    return click.option('--debug', '-d', is_flag=True, callback=callback,
```
Comment (xuanwang14) [inline databricks_cli/configure/config.py]: Probably we do not need "-d" here? It is okey to require the full option name for debug

**Comment by alinxie** [comment]
Added test and removed shorthand debug flag (-d)

**Comment by andrewmchen** [comment]
Thanks for the feature!

**Pull request merged**